main.go

```diff
@@ -1,33 +1,104 @@
 package main
 
 import (
-	"fmt"
-	"log"
+	"apify/actor/example/store"
+	"io"
+	"log/slog"
 	"net/http"
 	"os"
+	"strings"
+
+	"github.com/PuerkitoBio/goquery"
+	"github.com/geziyor/geziyor"
+	"github.com/geziyor/geziyor/client"
+	"github.com/geziyor/geziyor/export"
+)
+
+var (
+	xlog = slog.New(slog.NewTextHandler(os.Stdout, nil))
 )
 
+type (
+	KVStoreExporter map[string]any
+	AutherNText     struct {
+		Author string `json:"author"`
+		Text   string `json:"text"`
+	}
+)
+
+func (kv *KVStoreExporter) Export(parsedData chan any) error {
+	if len(*kv) == 0 {
+		*kv = make(KVStoreExporter)
+	}
+	data := []AutherNText{}
+	for d := range parsedData {
+		e := d.(map[string]interface{})
+		data = append(data, AutherNText{Author: e["author"].(string), Text: e["text"].(string)})
+	}
+	(*kv)["data"] = data
+
+	return nil
+}
+
+func RespBodyToStr(resp *http.Response) string {
+	b, err := io.ReadAll(resp.Body)
+	if err != nil {
+		xlog.Error("failed to read body", "error", err)
+	}
+	return string(b)
+}
+
 func main() {
-	log.Println("Example actor written in Go.")
-	resp, err := http.Get("https://example.com")
+	// Get Token and defualt KV store id
+	xlog.Info("Example actor written in Go.")
+
+	// Get default KV store
+	kv := store.KVStoreDefault()
+
+	// Get input
+	input, err := kv.Get("INPUT")
 	if err != nil {
-		log.Fatal(err)
+		xlog.Error("failed to get input from kv store", "error", err)
 		return
 	}
+	xlog.Info("input from kv store", "input", input)
+	url := input["url"].(string)
+	if strings.EqualFold(url, "") {
+		xlog.Error("no url in input", "url", url)
+	}
+	// scrape data
+	kvExporter := KVStoreExporter{}
+	scrape([]string{url}, &kvExporter)
 
-	default_kvs, token := os.Getenv("APIFY_DEFAULT_KEY_VALUE_STORE_ID"), os.Getenv("APIFY_TOKEN")
-	if default_kvs == "" || token == "" {
-		log.Fatal("Missing required env vars")
-		return
+	xlog.Info("saving scrapped data to kv store", "data", kvExporter)
+	if err := kv.Put("data", kvExporter); err != nil {
+		xlog.Error("error while add value to store", "data", kvExporter, "error", err)
 	}
-	client := http.Client{}
-	url := fmt.Sprintf("https://api.apify.com/v2/key-value-stores/%v/records/OUTPUT?token=%v", default_kvs, token)
-	req, _ := http.NewRequest(http.MethodPut, url, resp.Body)
-	req.Header.Set("Content-Type", "text/html; charset=utf-8")
-	_, err = client.Do(req)
-	if err != nil {
-		log.Fatal(err)
-		return
+	xlog.Info("actor is done")
+}
+
+func scrape(urls []string, exporter export.Exporter) {
+	g := geziyor.NewGeziyor(&geziyor.Options{
+		StartURLs: urls,
+		// StartRequestsFunc: func(g *geziyor.Geziyor) {
+		// 	g.GetRendered(urls[0], quotesParse)
+		// },
+		ParseFunc: quotesParse,
+		Exporters: []export.Exporter{&export.JSON{}, exporter},
+		// BrowserEndpoint: "ws://localhost:3000",
+	})
```

Comment:
Could you please remove the commented out code? thanks

Reply:
Yes sure